pithos/plugins/notification_icon.py

```diff
@@ -210,6 +222,92 @@ def on_disable(self):
         self.window.disconnect(self.delete_callback_handle)
         self.window.disconnect(self.state_callback_handle)
         self.window.disconnect(self.song_callback_handle)
+        self.preferences_dialog.disconnect(self.icon_changed_handler)
 
         # Pithos window needs to be reconnected to on_destro()
         self.window.connect('delete-event', self.window.on_destroy)
+
+
+class NotificationIconPluginPrefsDialog(Gtk.Dialog):
+    __gtype_name__ = 'NotificationIconPluginPrefsDialog'
+    __gsignals__ = {
+        'icon-changed': (GObject.SignalFlags.RUN_FIRST, None, (GObject.TYPE_STRING,)),
+    }
+
+    def __init__(self, parent, settings):
+        super().__init__(
+            _('Icon Type'),
+            parent,
+            0,
+            ('_Cancel', Gtk.ResponseType.CANCEL, '_Apply', Gtk.ResponseType.APPLY),
+            use_header_bar=1,
+        )
+        self.set_default_size(300, -1)
+        self.pithos_window = parent
+        self.settings = settings
+        self.set_resizable(False)
+        self.connect('response', self._on_response)
+
+        sub_title = Gtk.Label.new(_('Set the Notification Icon Type'))
+        sub_title.set_halign(Gtk.Align.CENTER)
+        self.icons_combo = Gtk.ComboBoxText.new()
+
+        icons = [
+            ('io.github.Pithos-tray', 'Full Color'),
+            ('io.github.Pithos-symbolic', 'Symbolic'),
+        ]
+
+        for icon in icons:
+            self.icons_combo.append(icon[0], icon[1])
+
+        self._reset_combo()
+        content_area = self.get_content_area()
+
+        content_area.add(sub_title)
+        content_area.add(self.icons_combo)
+        content_area.show_all()
+
+    def _reset_combo(self):
+        self.icons_combo.set_active_id(self.settings['data'])
+
+    def _on_response(self, dialog, response):
+        if response != Gtk.ResponseType.APPLY:
+            self.hide()
+            self._reset_combo()
+            return
+
+        setting = self.settings['data']
+        active_id = self.icons_combo.get_active_id()
+
+        if setting == active_id:
+            self.hide()
+            return
+
+        if active_id != 'io.github.Pithos-symbolic':
+            self.hide()
+            self.emit('icon-changed', active_id)
+            return
+
+        def on_dialog_response(dialog, response):
+            if response == Gtk.ResponseType.YES:
+                self.hide()
+                self.emit('icon-changed', active_id)
+            dialog.destroy()
+
+        message = (_(
```

Comment:
Do we need to have a warning really? Worst case it looks a little dumb but I think the user should understand it and nothing should break.